cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/CustomActionExecutor.java

```diff
@@ -64,10 +67,12 @@ private WorkflowAction createAction(BasicWorkflowContext context, InstantiatorFa
     Class<?> clz = Class.forName(context.getSpecification().getClassName(), true, classLoader);
     Preconditions.checkArgument(WorkflowAction.class.isAssignableFrom(clz), "%s is not a WorkflowAction.", clz);
     WorkflowAction action = instantiator.get(TypeToken.of((Class<? extends WorkflowAction>) clz)).create();
+    Metrics metrics = new ProgramUserMetrics(
+      context.getProgramMetrics().childContext(Constants.Metrics.Tag.NODE, action.getClass().getSimpleName()));
```

Comment:
what if same custom action is used multiple times in the Workflow? we should use `context.getSpecification().getName()` instead of `action.getClass().getSimpleName()`, right?